### ISSUE TITLE
Error: /etc/genders does not exist. After first cobbler sync

### ISSUE THREAD

**Issue by rbberger** [cobbler/cobbler#2169]
### Description
While the error message contains the fix, why is this necessary in the first place? Is there something missing in the RPM build? Or what causes `/etc/genders` to be necessary?

### Cobbler Version
```
Cobbler 3.0.1
  source: b'2b1c067c', b'Wed Oct 16 15:24:52 2019 -0400'
  build time: Thu Oct 17 01:46:46 2019
```

### Operating System

CentOS 8

### Output

```
[root@master tftpboot]# cobbler sync
task started: 2019-10-16_215532_sync
task started (id=Sync, time=Wed Oct 16 21:55:32 2019)
running python triggers from /var/lib/cobbler/triggers/task/sync/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/sync/pre/*
shell triggers finished successfully
running pre-sync triggers
cleaning trees
mkdir: /srv/tftpboot/pxelinux.cfg
mkdir: /srv/tftpboot/grub
mkdir: /srv/tftpboot/images
mkdir: /srv/tftpboot/ppc
mkdir: /srv/tftpboot/etc
removing: /srv/tftpboot/grub/images
copying bootloaders
running: rsync -rpt --copy-links --exclude=.cobbler_postun_cleanup /var/lib/cobbler/loaders/ /srv/tftpboot
received on stdout: 
received on stderr: 
running: rsync -rpt --copy-links --exclude=README.grubconfig /var/lib/cobbler/grub_config/ /srv/tftpboot
received on stdout: 
received on stderr: 
copying distros to tftpboot
copying images
generating PXE configuration files
generating PXE menu structure
cleaning link caches
running post-sync triggers
running python triggers from /var/lib/cobbler/triggers/sync/post/*
running python trigger cobbler.modules.sync_post_restart_services
running shell triggers from /var/lib/cobbler/triggers/sync/post/*
shell triggers finished successfully
running python triggers from /var/lib/cobbler/triggers/change/*
running python trigger cobbler.modules.scm_track
running python trigger cobbler.modules.managers.genders
Error: /etc/genders does not exist.
Please run: touch /etc/genders as root and try again.
Exception occured: <class 'cobbler.cexceptions.CX'>
Exception value: 'cobbler trigger failed: cobbler.modules.managers.genders'
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 84, in run
    rc = self._run(self)

  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 185, in runner
    self.remote.api.sync(self.options.get("verbose", False), logger=self.logger)

  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 719, in sync
    sync.run()

  File "/usr/lib/python3.6/site-packages/cobbler/actions/sync.py", line 137, in run
    utils.run_triggers(self.api, None, "/var/lib/cobbler/triggers/change/*", logger=self.logger)

  File "/usr/lib/python3.6/site-packages/cobbler/utils.py", line 839, in run_triggers
    raise CX("cobbler trigger failed: %s" % m.__name__)

!!! TASK FAILED !!!
```

**Comment by rbberger** [comment]
@SchoolGuy after looking at the code and the part that generates the error message, it seems like the error message is thrown before attempting to render the `/etc/genders` file.

cobbler/modules/managers/genders.py:94
```python
    # the file doesn't exist and for some reason the template engine
    # won't create it, so spit out an error and tell the user what to do.
    if(not os.path.isfile(settings_file)):
        logger.info("Error: " + settings_file + " does not exist.")
        logger.info("Please run: touch " + settings_file + " as root and try again.")
        return 1

    # <--- first rendering happens here? so why spit out the error before attempting it?
    write_genders_file(api._collection_mgr, profiles_genders, distros_genders, mgmtcls_genders)
```

It seems like a logic error to me. But I know nothing about this `genders` thing...

**Comment by watologo1** [comment]
This seem to come from api.settings.manage_genders

which is set to default 1 in a sub-config file:
cobbler/settings.py:    "manage_genders": [0, "bool"],
config/cobbler/settings.d/manage_genders.settings:# manage_genders - Bool to enable/disable managing an /etc/genders file for use with pdsh and others.
config/cobbler/settings.d/manage_genders.settings:manage_genders: 1

Setting this to default 0 should be just fine?

**Comment by SchoolGuy** [comment]
@watologo1 Yes this is the first part. Thanks for pointing this out.

@rbberger Yes I saw it but I want to dive deeper in the code to not accidently break something. Most probably this will happen on the weekend.

If I find what the `genders`-file and librarys are doing I will docuement it but there are quite some manpages so it can not be just a leaf thingy which is a gimmik I think this is part of the OS in some constellations... (Already saw the package for it in our OBS.)

**Comment by SchoolGuy** [comment]
Okay what the actual hell is genders? - Answer: https://computing.llnl.gov/linux/genders.html

**Comment by SchoolGuy** [comment]
Okay I wrote a small E-Mail to the original contributor of the genders-module with the wish of a little guide or docs contribution (I didn't see much sense in pinging him as these contributions were his only ever). Let's see if I get a reply.

**Comment by watologo1** [comment]
What I have heard is, that genders is a small file with some machine/node information which may be used by other applications inside a cloud.
So this is definitely not mandatory, so I would set default enabled to 0 (to avoid the bug Richard mentioned here. Once someone enables this, he should know what to provide to get a genders file created).
But as code to generate the file is there and it's tiny, I would just keep the rest.

**Comment by rbberger** [comment]
Fixed by c386e96
